lib/quantizeAttributes.js

```diff
@@ -0,0 +1,348 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+module.exports = quantizeAttributes;
+
+/**
+ *
+ * @param gltf
+ * @param options
+ *
+ * @returns glTF with quantized attributes
+ *
+ * @example
+ * // Quantize the positions of a simple mesh
+ * var gltf = {
+ *     "accessors": {
+ *         "accessor_0": {...}
+ *         "accessor_1": {...}
+ *     },
+ *     "meshes": {
+ *         "geometry": {
+ *             "name": "Mesh",
+ *             "primitives": [
+ *                 {
+ *                     "attributes": {
+ *                         "NORMAL": "accessor_0",
+ *                         "POSITION": "accessor_1"
+ *                     }
+ *                 }
+ *             ]
+ *         }
+ *     }
+ * };
+ * var options = {
+ *     "attributes": {
+ *         "geometry": {
+ *             "0": ["POSITION"]
+ *         }
+ *     }
+ * };
+ * quantizeAttributes(gltf, options);
+ */
+function quantizeAttributes(gltf, options) {
+    var accessors = gltf.accessors;
+    var meshes = gltf.meshes;
+    var extensionsUsed = gltf.extensionsUsed;
+    var accessorIds = {};
+    var attributes;
+    var attribute;
+    var accessor;
+    var meshId;
+    var primitives;
+    var primitive;
+    var i, j;
+    var index;
+
+    // Retrieve the accessors that should be quantized
+    var quantizeAll = true;
+    if (defined(options)) {
+        attributes = options.attributes;
```

Comment:
Also move this into its own function.  I think it will make this function flow really well.